src/main/java/com/commerce/backendserver/global/exception/ApiExceptionHandler.java

```diff
@@ -0,0 +1,34 @@
+package com.commerce.backendserver.global.exception;
+
+import com.commerce.backendserver.global.exception.error.ErrorCode;
+import com.commerce.backendserver.global.exception.error.ErrorResponse;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@Slf4j
+@RestControllerAdvice
+public class ApiExceptionHandler {
+
+  @ResponseStatus(HttpStatus.BAD_REQUEST)
+  @ExceptionHandler(MethodArgumentNotValidException.class)
+  public ErrorResponse handleMethodArgumentNotValidException(MethodArgumentNotValidException ex) {
+    log.error("", ex);
+    return ErrorResponse.of(HttpStatus.BAD_REQUEST, ex.getFieldError());
+  }
```

Comment:
`@ResponseStatus` 어노테이션으로 `BAD_REQUEST` 응답을 내려주는 핸들러인 것 같아요.
Line 18과 Line 22의 BAD_REQUEST 리턴 부분이 중복이라고 생각 되는데

```suggestion
  @ExceptionHandler(MethodArgumentNotValidException.class)
  public ErrorResponse handleMethodArgumentNotValidException(MethodArgumentNotValidException ex) {
    log.error("", ex);
    return ErrorResponse.of(HttpStatus.BAD_REQUEST, ex.getFieldError());
  }
```

불필요한 어노테이션을 제거하는 것도 좋을 것 같아요.

Reply:
이 부분은 ResponseEntity 로 응답을 안해서 따로 어노테이션이 필요해서 달았습니다.
해당 핸들러 메소드는 ErrorCode 를 요구하지 않기 때문에 이런식으로 처리했는데 공통적으로 ResponseEntity 로 처리하는걸 희망하시면 바꿀수 있습니다.

Reply:
아, 리턴타입이 ResponseEntity가 아니였군요! 밑에서 구현한 of 메소드를 활용해 에러를 리턴하는 편의메소드가 아주 멋져요! 좋아요!